textile/chat-features.textile

```diff
@@ -144,6 +144,10 @@ h4(#rooms-lifecycle-operations). Room Lifecycle Operations
 *** @(CHA-RL7a1)@ The @RETRY@ operation - an internal process.
 *** @(CHA-RL7a2)@ The @RELEASE@ operation.
 *** @(CHA-RL7a3)@ The @ATTACH@ and @DETACH@ operations have equal precedence.
+* @(CHA-RL9)@ Many operations in the Chat SDK will still attempt to proceed if the room is in an @ATTACHING@ status. However, their ability to proceed will depend on the subsequent status that the room takes. This specification point is defined here to avoid unnecessary repetition. It is testable via the points that refer to it.
+** @(CHA-RL9a)@ When the room status is @ATTACHING@ at the point of operation commencement, the caller will subscribe a one-time listener to the room status.
+** @(CHA-RL9b)@ If the next room status received by the caller is @ATTACHED@, the operation shall proceed as normal.
+** @(CHA-RL9c)@ If the next room status received is any other value, the operation must throw an error. The @ErrorInfo@ shall have the @RoomInInvalidState@ error code from the "chat-specific error codes":#error-codes, with a status code of @500@. The @cause@ field must contain any error associated with the room status change, if present.
```

Comment:
If it's any other value, like `DETACHED`, `SUSPENDED` or `FAILED` we are throwing error with code `RoomInInvalidState` = 102107.
Why don't we throw the same in the case of CHA-PR3f, CHA-PR3 or any other pre-op check that doesn't involve a status change

Reply:
In principle I think this is reasonable - tbf when we wrote the spec point, `RoomInInvalidState` wasn't an error yet.

Though the thing we need to be careful of is that `RoomInInvalidState` is supposed to be a 5xx status code, whereas what we really want for CHA-PR3f is more like a 4xx, because its the client doing something wrong, I'll have a think about it.

Reply:
Change made